datafusion/sql/src/parser.rs

```diff
@@ -257,6 +257,9 @@ fn ensure_not_set<T>(field: &Option<T>, name: &str) -> Result<(), ParserError> {
     Ok(())
 }
 
+// By default, allow expressions up to this deep before error
```

Comment:
Maybe we can say this is 2x the default limit of sqlparser so it is clear this is meant to *increase* the limit

```suggestion
// By default, allow expressions up to this deep before error
// (2x the default in sqlparser)
```